model_compression_toolkit/xquant/common/model_analyzer_utils.py

```diff
@@ -0,0 +1,101 @@
+#  Copyright 2024 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#  ==============================================================================
+from typing import Any, Dict, List, Tuple
+
+from abc import ABC, abstractmethod
+
+from model_compression_toolkit.xquant.logger import Logger
+
+
+class ModelAnalyzerUtils(ABC):
```

Comment:
maybe remove the "utils" part from the name, this is **the** "ModelAnalyzer", too many utils in this project